pkg/operator/operatorclient/interfaces.go

```diff
@@ -79,3 +87,97 @@ func (c *SecondarySchedulerClient) GetObjectMeta() (meta *metav1.ObjectMeta, err
 	}
 	return &instance.ObjectMeta, nil
 }
+
+func (c *SecondarySchedulerClient) ApplyOperatorSpec(ctx context.Context, fieldManager string, desiredConfiguration *applyconfiguration.OperatorSpecApplyConfiguration) error {
+	if desiredConfiguration == nil {
+		return fmt.Errorf("applyConfiguration must have a value")
+	}
+
+	jsonBytes, err := json.Marshal(desiredConfiguration)
```

Comment:
Why do you marshal/unmarshal in here?. Because `desiredConfiguration` and `operatorSpec` are same type.
Wouldn't this work;

```go
desiredSpec := &secondaryschedulerapplyconfiguration.SecondarySchedulerSpecApplyConfiguration{
		OperatorSpecApplyConfiguration: *desiredConfiguration,
	}
```

Reply:
Good point :) Thx